corehq/ex-submodules/casexml/apps/phone/tasks.py

```diff
@@ -37,7 +38,16 @@ def force_update_cleanliness_flags():
 def get_async_restore_payload(restore_config):
     """Process an async restore
     """
-    response = restore_config.generate_payload(async_task=current_task)
+    status = 'err'
+    try:
+        with restore_config.timing_context:
+            response = restore_config.generate_payload(async_task=current_task)
+        status = 200
+    except RestoreException:
+        status = 412
+        raise
+    finally:
+        record_restore_timing(restore_config, status)
```

Comment:
@millerdev , thinking about this a bit further, is there a way you can make this track how long the whole process took, from when the task was kicked off, to when a response was received? I'm worried that this still doesn't take into account the fact that celery might not run this task right away, and there is some overhead in there.